src/Utils/Block.ts

```diff
@@ -0,0 +1,183 @@
+		this.eventBus().emit(Block.EVENTS.FLOW_RENDER);
+	}
+
+	componentDidMount() {}
+
+	private _componentDidUpdate(oldProps: propsType, newProps: propsType) {
+		const response = this.componentDidUpdate(oldProps, newProps);
+		if (!response) {
+			return;
+		}
+		this._render();
+	}
+
+	componentDidUpdate(oldProps: propsType, newProps: propsType): boolean {
+		return oldProps !== newProps;
+	}
+
+	setProps = ({ ...nextProps }: propsType) => {
+		const previousProps = { ...this.props };
+		if (!nextProps) {
+			return;
+		}
+		Object.assign(this.props, nextProps);
+		this.eventBus().emit(Block.EVENTS.FLOW_CDU, previousProps, this.props);
+	};
+
+	get element() {
+		return this._element.firstElementChild;
+	}
+
+	private _render() {
+		const block = this.render();
+		this._removeEvents();
+		this._element.textContent = '';
+		if (block) {
+			this._element.appendChild(block);
+		}
+		this._addEvents();
+	}
+
+	compile = (tmpl: string, props: propsType) => {
+		const fragment = document.createElement('template');
+		const template = Handlebars.compile(tmpl);
+		const nodes = this._nodes;
+
+		Object.entries(props).forEach(([propName, prop], index) => {
+			if (prop instanceof Block) {
+				nodes[index] = prop;
+				props[propName] = `<div id='${index}'></div>`;
+			}
+		});
+
+		fragment.innerHTML = template(props);
+
+		Object.entries(nodes).forEach(([id, component]: [string, Block]) => {
+			const templateEl = fragment.content.getElementById(`${id}`);
+
+			if (!templateEl) {
+				return;
+			}
+
+			templateEl.replaceWith(component.render());
+		});
+
+		if (props.events) {
+			this._addEvents(fragment.content.firstElementChild);
+		}
+
+		return fragment.content.firstElementChild;
+	};
+
+	getContent() {
+		return this.element as Node;
+	}
+
+	private _addEvents(element?: any) {
+		const { events } = this.props;
+		if (events) {
+			Object.entries(events).forEach(([event, handler]) => {
+				if (element) {
+					element.addEventListener(event, handler);
+				}
+				this.element?.addEventListener(event, handler);
+			});
+		}
+	}
+
+	private _removeEvents(element?: any) {
+		const { events } = this.props;
+		if (events) {
+			Object.entries(events).forEach(([event, handler]) => {
+				if (element) {
+					element.removeEventListener(event, handler);
+				}
+				this.element?.removeEventListener(event, handler);
+			});
+		}
+	}
+
+	private _makePropsProxy = (props: propsType) => {
+		// eslint-disable-next-line @typescript-eslint/no-this-alias
+		const self = this;
+		return new Proxy(props, {
+			get(target, prop: string) {
+				if (prop.indexOf('_') === 0) {
+					throw new Error('Нет доступа');
+				}
+				const value = target[prop];
+				return typeof value === 'function' ? value.bind(target) : value;
+			},
+			set(target, prop: string, value) {
+				if (prop.indexOf('_') === 0) {
+					throw new Error('Нет доступа');
+				}
+				target[prop] = value;
+				self.eventBus().emit(Block.EVENTS.FLOW_CDU, { ...target }, target);
```

Comment:
Можно использовать стрелочные функции, тогда не придется прибегать к замыканию
```suggestion
			set: (target, prop: string, value) => {
				if (prop.indexOf('_') === 0) {
					throw new Error('Нет доступа');
				}
				target[prop] = value;
				this.eventBus().emit(Block.EVENTS.FLOW_CDU, { ...target }, target);
```